nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -282,8 +290,8 @@ def reduce_overlap_gradients(self):
                 p for p in self._optimizer.parameters() if not getattr(p, '_disable_overlap_grad_sync', False)
             )
 
-    def on_train_batch_end(self, outputs, batch, batch_idx: int, unused: Optional[int] = 0) -> None:
-        super().on_train_batch_end(outputs, batch, batch_idx)
+    def on_train_batch_end(self, outputs, dataloader_iter: Any, batch_idx: int, unused: Optional[int] = 0) -> None:
+        super().on_train_batch_end(outputs, dataloader_iter, batch_idx)
```

Comment:
This mean we have to update all nemo megatron models to use the dataloader_iter? If we want to first do this for GPT only can we move this to MegatronGPTModel?